src/components/src/main/java/org/apache/jmeter/extractor/json/jmespath/JMESPathExtractor.java

```diff
@@ -0,0 +1,242 @@
+                    vars.put(refName, defaultValue);
+                    vars.put(refName + REF_MATCH_NR, "0"); //$NON-NLS-1$
+                    if (matchNumber < 0) {
+                        log.debug("No value extracted, storing empty in: {}", refName);
+                    }
+                } else {
+                    List<String> resultList = splitJson(result);
+                    // if more than one value extracted, suffix with "_index"
+                    if (resultList.size() > 1) {
+                        if (matchNumber < 0) {
+                            // Extract all
+                            int index = 1;
+                            for (String extractedString : resultList) {
+                                vars.put(refName + "_" + index, extractedString); // $NON-NLS-1$
+                                index++;
+                            }
+                        } else if (matchNumber == 0) {
+                            // Random extraction
+                            int matchSize = resultList.size();
+                            int matchNr = JMeterUtils.getRandomInt(matchSize);
+                            placeObjectIntoVars(vars, refName, resultList, matchNr);
+                        } else {
+                            // extract at position
+                            if (matchNumber > resultList.size()) {
+                                if (log.isDebugEnabled()) {
+                                    log.debug(
+                                            "matchNumber({}) exceeds number of items found({}), default value will be used",
+                                            matchNumber, resultList.size());
+                                }
+                                vars.put(refName, defaultValue);
+                            } else {
+                                placeObjectIntoVars(vars, refName, resultList, matchNumber - 1);
+                            }
+                        }
+                    } else {
+                        // else just one value extracted
+                        String suffix = (matchNumber < 0) ? "_1" : "";
+                        placeObjectIntoVars(vars, refName + suffix, resultList, 0);
+                    }
+                    vars.put(refName + REF_MATCH_NR, Integer.toString(resultList.size()));
+                }
+            }
+        } catch (Exception e) {
+            // if something wrong, default value added
+            if (log.isDebugEnabled()) {
+                log.debug("Error processing JSON content in {}, message: {}", getName(), e.getLocalizedMessage(), e);
+            } else {
+                log.debug("Error processing JSON content in {}, message: {}", getName(), e.getLocalizedMessage());
+            }
+            vars.put(refName, defaultValue);
+        }
+    }
+
+    public List<String> splitJson(JsonNode jsonNode) throws JsonParseException, JsonMappingException, IOException {
```

Comment:
The exceptions are subclasses of `IOException`. Do we  really need to specify them here?